mm2src/mm2_main/src/lp_swap/saved_swap.rs

```diff
@@ -259,10 +262,87 @@ mod native_impl {
 #[cfg(target_arch = "wasm32")]
 mod wasm_impl {
     use super::*;
-    use crate::mm2::lp_swap::swap_wasm_db::{DbTransactionError, InitDbError, SavedSwapTable};
-    use crate::mm2::lp_swap::SwapsContext;
+    use crate::mm2::lp_swap::swap_wasm_db::{DbTransactionError, InitDbError, MySwapsFiltersTable, SavedSwapTable,
+                                            SwapsMigrationTable};
+    use crate::mm2::lp_swap::{SwapsContext, LEGACY_SWAP_TYPE};
+    use bytes::Buf;
+    use common::log::{info, warn};
+    use mm2_db::indexed_db::cursor_prelude::CursorError;
+    use mm2_db::indexed_db::DbTable;
     use serde_json as json;
 
+    pub(super) async fn get_current_migration(
+        migration_table: &DbTable<'_, SwapsMigrationTable>,
+    ) -> MmResult<u32, SavedSwapError> {
+        let migrations = migration_table
+            .cursor_builder()
+            .bound("migration", 0, u32::MAX)
+            .reverse()
+            .open_cursor("migration")
+            .await?
+            // TODO refactor when "closure invoked recursively or after being dropped" is fixed
+            .collect()
+            .await?;
+
+        Ok(migrations.first().map(|(_, m)| m.migration).unwrap_or_default())
```

Comment:
I have a fix for for this issue now..So I think this pr can be hold till https://github.com/KomodoPlatform/komodo-defi-framework/pull/2028 is merged then you can query first directly instead of collecting all.

@artemii235 @shamardy

Reply:
Perfect, thanks!